packages/react/src/components/table/Table.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+// import core base styles
+import 'hds-core';
+import classNames from '../../utils/classNames';
+import styles from './Table.module.scss';
+import { TableBody } from './TableBody';
+import { HeaderRow } from './HeaderRow';
+import { VerticalHeaderColGroup } from './VerticalHeaderColGroup';
+import { SortingHeaderCell } from './SortingHeaderCell';
+
+export type TableProps = React.ComponentPropsWithoutRef<'table'> & {
+  children: React.ReactNode;
+  variant?: 'dark' | 'light';
+  dense?: boolean;
+  zebra?: boolean;
+  verticalLines?: boolean;
+  customThemeClass?: string;
+};
+
+export const Table = ({
+  children,
+  variant = 'dark',
+  dense = false,
+  zebra = false,
+  verticalLines = false,
+  customThemeClass,
+  ...rest
+}: TableProps) => {
+  return (
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+    <div tabIndex={0} className={styles.container}>
+      <table
+        className={classNames(
+          styles.table,
+          variant === 'dark' ? styles.dark : styles.light,
+          dense && styles.dense,
+          zebra && styles.zebra,
+          verticalLines && styles.verticalLines,
+          customThemeClass,
+        )}
+        {...rest}
+      >
+        {children}
+      </table>
+    </div>
+  );
+};
+
+Table.TableBody = TableBody;
```

Comment:
Do we need/want these sub-components to be public? At some point we had the idea of people being able to compose their own table. But it was when we thought that more complex features like sorting and checkbox selection would not be reasonable to do otherwise than using composition and story examples. Now that we actually managed to implement them directly from DataTable api, Im not sure if we need/benefit that much from exposing these? It does increase the complexity and a bit and then makes it harder to change the sub-components later on without breaking changes 🤔

Reply:
It would help to decide between DataTable and/or Table composition if we knew how tables are used in Helsinki Kaupunki projects. At first, it would feel easier to start with a concise API, but I guess it could be verified. And I'm a bit worried that we can not make breaking changes to new components because then we have to get them right on the first try, and then they are locked until the next major release 🤔

Reply:
Nothing stops us from making every release a major release. Of course it's ideal if there isn't breaking changes all the time, but as long as the version numbers reflect the changes correctly.

Reply:
We decided to publish only Table now (former DataTable), and then publish sub-components later on if they are needed / wanted.